server/src/test/java/org/elasticsearch/index/engine/CombinedDeletionPolicyTests.java

```diff
@@ -244,6 +244,33 @@ public void testKeepOnlyStartingCommitOnInit() throws Exception {
             equalTo(Long.parseLong(startingCommit.getUserData().get(Translog.TRANSLOG_GENERATION_KEY))));
     }
 
+    public void testCheckUnreferencedCommits() throws Exception {
+        final AtomicLong globalCheckpoint = new AtomicLong(SequenceNumbers.UNASSIGNED_SEQ_NO);
+        final UUID translogUUID = UUID.randomUUID();
+        CombinedDeletionPolicy indexPolicy = new CombinedDeletionPolicy(
+            OPEN_INDEX_AND_TRANSLOG, createTranslogDeletionPolicy(), globalCheckpoint::get, null);
+        final List<IndexCommit> commitList = new ArrayList<>();
+        int totalCommits = between(2, 20);
+        long lastMaxSeqNo = between(1, 1000);
+        long lastTranslogGen = between(1, 50);
+        for (int i = 0; i < totalCommits; i++) {
+            lastMaxSeqNo += between(1, 10000);
+            lastTranslogGen += between(1, 100);
+            commitList.add(mockIndexCommit(lastMaxSeqNo, translogUUID, lastTranslogGen));
+        }
+        IndexCommit safeCommit = randomFrom(commitList);
+        globalCheckpoint.set(Long.parseLong(safeCommit.getUserData().get(SequenceNumbers.MAX_SEQ_NO)));
+        indexPolicy.onCommit(commitList);
+        globalCheckpoint.set(randomLongBetween(globalCheckpoint.get(), lastMaxSeqNo)); // Advanced not enough
+        assertThat(indexPolicy.hasUnreferencedCommits(), equalTo(false));
+        globalCheckpoint.set(randomLongBetween(lastMaxSeqNo, Long.MAX_VALUE));
+        if (safeCommit == commitList.get(commitList.size() - 1)) {
+            assertThat(indexPolicy.hasUnreferencedCommits(), equalTo(false)); // Keeping a single commit
+        } else {
+            assertThat(indexPolicy.hasUnreferencedCommits(), equalTo(true));
```

Comment:
can we run another on commit and check that we just have one commit left and that the indexPolicy. hasUnreferencedCommits now returns true?